.github/workflows/niceplots.yml

```diff
@@ -29,14 +32,17 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip wheel
+        python -m pip install flake8 black 
+        python -m pip install numpy==${{ matrix.numpy-version }} 
+        python -m pip install matplotlib==${{ matrix.mpl-version }}
         pip install .
-        python -m pip install flake8 black numpy matplotlib
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
     - name: Lint with flake8
+      if: matrix.numpy-version == '1.16.6' && matrix.mpl-version == '3.2.*'
```

Comment:
I would prefer if these are separate (i.e. parallel) jobs instead of an additional step

Reply:
Right, seems to be skipping linting/formatting on all the other combinations, so if possible a dedicated job would make sense.

Reply:
Yes I agree, I just hadn't figured out how to do it yet, plus I wanted to run black once for each python version

Reply:
Yeah I can just push up some changes to the yaml file